### PR TITLE
move history details style to css

### PR DESCRIPTION
**Pull request by le-jeu**

### CODE REVIEW

File: core/code/portal_detail_display_tools.js [hunk 1]
```diff
@@ -4,17 +4,17 @@
 
 window.getPortalHistoryDetails = function (d) {
   if (!d.history) {
-    return '<div style="text-align: center; color: #ffce00">History missing</div>';
+    return '<div id="history_details">History missing</div>';
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]: May be add some class for missing history?

File: core/code/portal_detail_display_tools.js [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
   var scoutControlled = d.history.scoutControlled;
-  var colors = {true:'#03fe03', false:'#ff4a4a'};
+  var className = {true:'done', false:''};
 
-  return '<div style="text-align: center; color: #ffce00">History: '
-  + '<span id="history_visited" style="color:'+ colors[visited] + '">visited</span> | '
-  + '<span id="history_captured" style="color:'+ colors[captured] + '">captured</span> | '
-  + '<span id="history_scout_controlled" style="color:' + colors[scoutControlled] + '">scout controlled</span>'
+  return '<div id="history_details">History: '
+  + '<span id="history_visited" class="'+ className[visited] + '">visited</span> | '
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]: May be shorten ids?
`visited`
`captured`
`scout-controlled`

File: core/code/portal_detail_display_tools.js [hunk 3]
```diff
@@ -4,17 +4,17 @@
...
 window.getPortalHistoryDetails = function (d) {
   if (!d.history) {
-    return '<div style="text-align: center; color: #ffce00">History missing</div>';
+    return '<div id="history_details">History missing</div>';
   }
   var visited = d.history.visited;
   var captured = d.history.captured;
   var scoutControlled = d.history.scoutControlled;
-  var colors = {true:'#03fe03', false:'#ff4a4a'};
+  var className = {true:'done', false:''};
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]:
```suggestion
  var className = {true:'class="done" ', false:''};
```

File: core/code/portal_detail_display_tools.js [hunk 4]
```diff
@@ -4,17 +4,17 @@
...
 
-  return '<div style="text-align: center; color: #ffce00">History: '
-  + '<span id="history_visited" style="color:'+ colors[visited] + '">visited</span> | '
-  + '<span id="history_captured" style="color:'+ colors[captured] + '">captured</span> | '
-  + '<span id="history_scout_controlled" style="color:' + colors[scoutControlled] + '">scout controlled</span>'
+  return '<div id="history_details">History: '
+  + '<span id="history_visited" class="'+ className[visited] + '">visited</span> | '
+  + '<span id="history_captured" class="'+ className[captured] + '">captured</span> | '
+  + '<span id="history_scout_controlled" class="' + className[scoutControlled] + '">scout controlled</span>'
   + '</div>';
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]: Perhaps this could be done with https://leafletjs.com/reference-1.7.1.html#util-template

**Review by johnd0e — CHANGES_REQUESTED** [review]

File: core/code/portal_detail_display_tools.js [hunk 5]
```diff
@@ -4,18 +4,18 @@
...
-
-  return '<div style="text-align: center; color: #ffce00">History: '
-  + '<span id="history_visited" style="color:'+ colors[visited] + '">visited</span> | '
-  + '<span id="history_captured" style="color:'+ colors[captured] + '">captured</span> | '
-  + '<span id="history_scout_controlled" style="color:' + colors[scoutControlled] + '">scout controlled</span>'
-  + '</div>';
+  var classParts = {}
+  classParts.visited = (d.history.visited) ? 'class="done"' : "";
+  classParts.captured = (d.history.captured) ? 'class="done"' : "";
+  classParts.scoutControlled = (d.history.scoutControlled) ? 'class="done"' : "";
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]:
```suggestion
  ['visited', 'captured', 'scoutControlled'].forEach(function (prop) {
    classParts[prop] = d.history[prop] ? 'class="done"' : '';
  });
```

File: core/style.css [hunk 6]
```diff
@@ -836,6 +836,23 @@ h3.title {
...
+}
+
+#historydetails .missing {
+}
+
+#historydetails span {
+  color: #ff4a4a;
+}
+
+#historydetails span.done {
```
Comment (johnd0e) [inline core/style.css]: I wonder which classname fit best: done, completed, checked-in, ...

**Review by johnd0e — CHANGES_REQUESTED** [review]

File: core/code/portal_detail_display_tools.js [hunk 7]
```diff
@@ -4,18 +4,18 @@
...
-  var captured = d.history.captured;
-  var scoutControlled = d.history.scoutControlled;
-  var colors = {true:'#03fe03', false:'#ff4a4a'};
-
-  return '<div style="text-align: center; color: #ffce00">History: '
-  + '<span id="history_visited" style="color:'+ colors[visited] + '">visited</span> | '
-  + '<span id="history_captured" style="color:'+ colors[captured] + '">captured</span> | '
-  + '<span id="history_scout_controlled" style="color:' + colors[scoutControlled] + '">scout controlled</span>'
-  + '</div>';
+  var classParts = {}
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]:
```suggestion
  var classParts = {};
```

File: core/style.css (hunk 6 above)
Comment (le-jeu) [reply core/style.css]: I don't know, completed is better than done
we can also use checkbox

File: core/code/portal_detail_display_tools.js [hunk 8]
```diff
@@ -4,18 +4,18 @@
...
-  var colors = {true:'#03fe03', false:'#ff4a4a'};
-
-  return '<div style="text-align: center; color: #ffce00">History: '
-  + '<span id="history_visited" style="color:'+ colors[visited] + '">visited</span> | '
-  + '<span id="history_captured" style="color:'+ colors[captured] + '">captured</span> | '
-  + '<span id="history_scout_controlled" style="color:' + colors[scoutControlled] + '">scout controlled</span>'
-  + '</div>';
+  var classParts = {};
+  ['visited', 'captured', 'scoutControlled'].forEach(function (k) {
+    classParts[k] = (d.history[k]) ? 'class="completed"' : "";
```
Comment (johnd0e) [inline core/code/portal_detail_display_tools.js]: Why do we need brackets here?

Comment (le-jeu) [reply core/code/portal_detail_display_tools.js]: *parentheses ? we don't

File: core/style.css (hunk 6 above)
Comment (johnd0e) [reply core/style.css]: That's why `checked` (or `checked-in`) makes sense too.

File: core/style.css [hunk 9]
```diff
@@ -836,6 +836,23 @@ h3.title {
...
+#historydetails {
+  text-align: center;
+  color: #ffce00;
+}
+
+#historydetails .missing {
+}
+
+#historydetails span {
+  color: #ff4a4a;
```
Comment (johnd0e) [inline core/style.css:849]: I'm not sure about exact values.
We should use the same as in other iitc styles, and I am unable to find `#ff4a4a`/`#03fe03`.

**Pull request merged**